### PR TITLE
feat: add homework 3 files

### PR DESCRIPTION
**Pull request by annavladimab**

### CODE REVIEW

File: Homeworks/homework_3/home_work_3_33.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        start = 0
+    if type(stop) == str:
+        stop = sequence.index(stop)
+        start = sequence.index(start)
+    if start > stop:
+        sequence = sequence[stop + 1:start + 1]
+        step = -step
+        sequence.reverse()
+        return sequence[::step]
+    return sequence[start:stop][::step]
```
Comment (alpakhomov) [inline Homeworks/homework_3/home_work_3_33.py:53]: I think it can just be `sequence[start:stop:step]`

**Review by alpakhomov — CHANGES_REQUESTED** [review]
Please fix all the comments + you are missing fourth task in homework 3

File: Homeworks/homework_2/hw2.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Input: {1: ["A", "E"], 2: ["D", "G"]}
+Output: {"a": 1, "d": 2, "e": 1, "g": 2}
+"""
+
+
+from typing import Dict, List
+
+def transform(legacy_data: Dict[int, List[str]]) -> Dict[str, int]:
+    ungroup = {value.lower(): key for key, values in legacy_data.items() for value in values}
+    return dict(sorted(ungroup.items()))
```
[the remark this answers is not in the document]
Comment (annavladimab) [reply Homeworks/homework_2/hw2.py]: issue was fixed
Thank you!

File: Homeworks/homework_2/hw2.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    """
+    ungroup = {}
+    for key, values in legacy_data.items():
+        for value in values:
+            ungroup[value.lower()] = key
+    sorted_values =  sorted(ungroup.items())
+    sorted_item =  dict(sorted_values)
+    return sorted_item
+"""
```
[the remark this answers is not in the document]
Comment (annavladimab) [reply Homeworks/homework_2/hw2.py]: formatting was changed

File: Homeworks/homework_3/home_work_2_33.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+assert val_1 is val_2
+"""
+from collections.abc import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cashe = {}
+    def inner_cashe_func(a, b):
```
[the remark this answers is not in the document]
Comment (annavladimab) [reply Homeworks/homework_3/home_work_2_33.py]: fixed

File: Homeworks/homework_3/home_work_3_33.py (hunk 1 above)
Comment (annavladimab) [reply Homeworks/homework_3/home_work_3_33.py]: fixed

**Pull request merged**